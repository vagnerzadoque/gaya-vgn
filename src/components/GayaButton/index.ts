import GayaButton from './GayaButton'

export { GayaButton }
export { GayaButtonProps } from './GayaButton.props'
export default GayaButton
