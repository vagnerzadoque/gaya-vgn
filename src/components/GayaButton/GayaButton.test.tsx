/* eslint-disable max-lines-per-function */
import * as React from 'react'
import { fireEvent } from '@testing-library/react'
import GayaButton from '.'
import renderWithTheme from '../../helpers/renderWithTheme'
import { GayaButtonProps } from './GayaButton.props'

jest.mock('../Ripple/Ripple')

const buttonProps: GayaButtonProps = {
  onClick: () => '',
  children: 'button'
}

describe('Button component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should call onClick', () => {
    const onClickMock = jest.fn()
    const { component: { getByTestId } } = renderWithTheme(<GayaButton {...buttonProps} onClick={onClickMock} testID="btn-enable" />)

    fireEvent.click(getByTestId('btn-enable'))

    expect(onClickMock).toHaveBeenCalled()
  })

  it('should not call onClick when is disabled', () => {
    const onClickMock = jest.fn()
    const { component: { getByTestId } } = renderWithTheme(<GayaButton {...buttonProps} disabled onClick={onClickMock} testID="btn-disabled" />)

    fireEvent.click(getByTestId('btn-disabled'))

    expect(getByTestId('btn-disabled')).toBeDisabled()
    expect(onClickMock).not.toHaveBeenCalled()
  })

  it('should render correctly when the size is semi', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} size="semi" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the size is medium', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} size="medium" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the fullWidth is true', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} fullWidth />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when the variant is contained and disabled is true', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} disabled testID="btn-disabled" />)

    expect(component.getByTestId('btn-disabled')).toBeDisabled()
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is outlined', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} variant="outlined" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is text', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} variant="text" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when the variant is text and is disabled', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} variant="text" disabled />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when the variant is outlined and is disabled', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} variant="outlined" disabled />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render icon to the right as default when the showIcon is true', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} showIcon iconName="outlined-default-mockup" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render icon to the left when the showIcon is true and iconPosition is left', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} showIcon iconPosition="left" iconName="outlined-default-mockup" />)

    expect(component.getByTestId('icon-outlined-default-mockup')).toBeTruthy()
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with the given icon name', () => {
    const { styles, component } = renderWithTheme(<GayaButton {...buttonProps} showIcon iconName="filled-action-check" />)

    expect(component.getByTestId('icon-filled-action-check')).toBeTruthy()
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
