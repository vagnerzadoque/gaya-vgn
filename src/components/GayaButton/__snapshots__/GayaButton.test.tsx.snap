// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly when the fullWidth is true 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: 100%;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the size is medium 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 48px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the size is semi 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 32px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is contained and disabled is true 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        aria-disabled="true"
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        data-testid="btn-disabled"
        disabled=""
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is outlined 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #FFFFFF00;
  border: 1px solid #f4ab34;
  border-radius: 4px;
  color: #363636;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #e1e1e1;
  border: 1px solid #f6b854;
  color: #111111;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #F8C675;
  color: #111111;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #F8C675;
  color: #111111;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is outlined and is disabled 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #FFFFFF00;
  border: 1px solid #f4ab34;
  border-radius: 4px;
  color: #363636;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #e1e1e1;
  border: 1px solid #f6b854;
  color: #111111;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #F8C675;
  color: #111111;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #F8C675;
  color: #111111;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        aria-disabled="true"
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        disabled=""
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is text 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #FFFFFF00;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #f4ab34;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #e1e1e1;
  border: 1px solid #FFFFFF00;
  color: #111111;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #FFFFFF00;
  color: #111111;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #FFFFFF00;
  color: #111111;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly when the variant is text and is disabled 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #FFFFFF00;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #f4ab34;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #e1e1e1;
  border: 1px solid #FFFFFF00;
  color: #111111;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #FFFFFF00;
  color: #111111;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #c9c9c9;
  border: 1px solid #FFFFFF00;
  color: #111111;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        aria-disabled="true"
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        disabled=""
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly with default props 1`] = `
[
  ".gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render correctly with the given icon name 1`] = `
[
  ".icon-0-2-7 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-8 {
  font-size: 24px;
}
.gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  margin-right: 8px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
          <i
            aria-hidden="true"
            class=" icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-filled-action-check Icon-gaya"
            data-testid="icon-filled-action-check"
            role="img"
          />
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render icon to the left when the showIcon is true and iconPosition is left 1`] = `
[
  ".icon-0-2-7 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-8 {
  font-size: 24px;
}
.gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row-reverse;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  margin-left: 8px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
          <i
            aria-hidden="true"
            class=" icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
            data-testid="icon-outlined-default-mockup"
            role="img"
          />
        </div>
      </button>
    </div>
  </div>,
]
`;

exports[`Button component should render icon to the right as default when the showIcon is true 1`] = `
[
  ".icon-0-2-7 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-8 {
  font-size: 24px;
}
.gayaButton-0-2-1 {
  cursor: pointer;
  outline: 0;
  overflow: hidden;
  position: relative;
  box-shadow: none;
}
.gayaButton-0-2-1:disabled {
  border: none;
  cursor: default;
  box-shadow: none;
}
.gayaButton-0-2-1:hover:not([disabled]) {}
.gayaButton-0-2-1:focus-visible:not([disabled]) {}
.gayaButton-0-2-1:active:not([disabled]) {}
.labelContainer-0-2-2 {
  display: flex;
  align-items: center;
  justify-content: center;
}
.label-0-2-3 {
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  user-select: none;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.gayaButton-d0-0-2-4 {
  background-color: #f4ab34;
  border: 1px solid #FFFFFF00;
  border-radius: 4px;
  color: #333333;
  height: 40px;
  padding-left: 8px;
  padding-right: 8px;
  width: auto;
}
.gayaButton-d0-0-2-4:disabled {
  background-color: #a4a4a4;
  color: #363636;
}
.gayaButton-d0-0-2-4:hover:not([disabled]) {
  background-color: #f6b854;
  border: 1px solid #FFFFFF00;
  color: #333333;
}
.gayaButton-d0-0-2-4:focus-visible:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.gayaButton-d0-0-2-4:active:not([disabled]) {
  background-color: #F8C675;
  border: 1px solid #FFFFFF00;
  color: #333333;
  opacity: 0.64;
}
.labelContainer-d1-0-2-5 {
  flex-direction: row;
}
.label-d2-0-2-6 {
  letter-spacing: 1.23px;
  margin-right: 8px;
  text-transform: uppercase;
}",
  <div>
    <div
      data-testid="ripple"
    >
      <button
        class="gayaButton-0-2-1 gayaButton-d0-0-2-4  Gaya-Button"
        type="button"
      >
        <div
          class="labelContainer-0-2-2 labelContainer-d1-0-2-5"
        >
          <span
            class="label-0-2-3 label-d2-0-2-6"
          >
            button
          </span>
          <i
            aria-hidden="true"
            class=" icon-0-2-7 icon-d0-0-2-8 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
            data-testid="icon-outlined-default-mockup"
            role="img"
          />
        </div>
      </button>
    </div>
  </div>,
]
`;
